llama-index-core/llama_index/core/indices/knowledge_graph/base.py

```diff
@@ -240,6 +241,19 @@ def _insert(self, nodes: Sequence[BaseNode], **insert_kwargs: Any) -> None:
                     rel_embedding = self._embed_model.get_text_embedding(triplet_str)
                     self._index_struct.add_to_embedding_dict(triplet_str, rel_embedding)
 
+    def insert(self, document: Document, **insert_kwargs: Any) -> None:
+        """Insert a document."""
+        with self._service_context.callback_manager.as_trace("insert_document"):
+            for doc in document:
+                self._docstore.set_document_hash(doc.get_doc_id(), doc.hash)
+            nodes = run_transformations(
+                document,
+                self._service_context.transformations,
```

Comment:
`ServiceContext` is deprecated. New features/capabilities should refrain from using it. Moreover, these shouldn't these transformation be stored already in `self._transformations`?